File: articles/mathematical-modeling-formal-representation-complex-systems/go/logistic_model.go
package main

import "fmt"

func simulateLogistic(initialState float64, growthRate float64, carryingCapacity float64, timeSteps int) []float64 {
	state := make([]float64, timeSteps)
	state[0] = initialState

	for t := 1; t < timeSteps; t++ {
		state[t] = state[t-1] + growthRate*state[t-1]*(1.0-state[t-1]/carryingCapacity)
	}

	return state
}

func main() {
	state := simulateLogistic(10.0, 0.18, 100.0, 80)

	fmt.Println("Final state:")
	fmt.Printf("%.3f\n", state[len(state)-1])
}

File: articles/mathematical-modeling-formal-representation-complex-systems/cpp/logistic_model.cpp
#include <iostream>
#include <vector>

std::vector<double> simulate_logistic(double initial_state, double growth_rate, double carrying_capacity, int time_steps) {
    std::vector<double> state(time_steps, 0.0);
    state[0] = initial_state;

    for (int t = 1; t < time_steps; ++t) {
        state[t] = state[t - 1] + growth_rate * state[t - 1] * (1.0 - state[t - 1] / carrying_capacity);
    }

    return state;
}

int main() {
    auto state = simulate_logistic(10.0, 0.18, 100.0, 80);

    std::cout << "Final state: " << state.back() << "\n";

    return 0;
}
